client/cmd.go

```diff
@@ -92,6 +92,11 @@ func ReadPersistentCommandFlags(clientCtx Context, flagSet *pflag.FlagSet) (Cont
 		output, _ := flagSet.GetString(cli.OutputFlag)
 		clientCtx = clientCtx.WithOutputFormat(output)
 	}
+	
+	if clientCtx.HomeDir == "" || flagSet.Changed(flags.FlagHome) {
```

Comment:
I think this is the correct approach and we probably should remove `ReadHomeFlag`.